src/main/java/edu/utdallas/davisbase/compiler/Compiler.java

```diff
@@ -366,6 +373,38 @@ public boolean checkIsNotNull(List<String> columnSpecs) {
     return false;
   }
 
+  /**
+   * @param columnSpecs column constraints for give column
+   * @return whether column is UNIQUE
+   */
+  public boolean checkIsUnique(List<String> columnSpecs) {
+    if (null != columnSpecs) {
+      if(columnSpecs.contains("UNIQUE")){
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @param columnSpecs column constraints for give column
+   * @return whether column is PRIMARY KEY
+   */
+  public boolean checkIsPrimaryKey(List<String> columnSpecs, Index index, String columnName) {
```

Comment:
This isn't a change request, just a comment.

If you ever have to check if a sublist exists within a list, rather than coding up the algorithm by hand, just use [`Collections#indexOfSubList(List<?> source, List<?> target)`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#indexOfSubList-java.util.List-java.util.List-)